src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlBulkCopy.cs

```diff
@@ -1518,10 +1518,14 @@ private object ConvertValue(object value, _SqlMetaData metadata, bool isNull, re
                         value = SqlParameter.CoerceValue(value, mt, out coercedToDataFeed, out typeChanged, false);
                         if (!coercedToDataFeed)
                         {   // We do not need to test for TextDataFeed as it is only assigned to (N)VARCHAR(MAX)
-                            int len = ((isSqlType) && (!typeChanged)) ? ((SqlString)value).Value.Length : ((string)value).Length;
-                            if (len > length / 2)
+                            string str = ((isSqlType) && (!typeChanged)) ? ((SqlString)value).Value : ((string)value);
+                            if (str.Length > length / 2)
```

Comment:
You might want to change this to include an IsNullOrEmpty check, there's nothing to guarantee that the str value is not null.

Reply:
It was carelessly accessing `.Length` before as well, so either it was broken before as well or it's somehow safe

Reply:
Actually, it is checked against here https://github.com/dotnet/SqlClient/blob/c8b60fc389ab01b7affeed673b634993d3179bfc/src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlBulkCopy.cs#L1414-L1421

Reply:
If clr null was passed in but the isNull parameter was false that code would return the null, and yes if it was doing that before I'd say it was careless. But since you're already improving it why not add the null check to improve this bit as well?

Reply:
IMO this would be a bug, in particular since the SQL Server column could be `NOT NULL` so just letting `null` slip through would only help along to have the bug crawl deeper into the code before it blows up.

Reply:
I would say it's not part of this PR, so rather let it be as it is.
L1414 `isNull` check should be revisited for this one if there's a possibility of sweeping in Null `value` with `isNull=false`.